streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -144,6 +144,13 @@ class StreamingContext private[streaming] (
     }
   }
 
+  if (sc.conf.contains("spark.cores.max")) {
+    val totalCores = sc.conf.getInt("spark.cores.max", 1)
```

Comment:
@jerryshao spark.cores.max will be returned only when conf contain it.
Here is the biggest possibility to make a judgment.